lib/vector-buffers/src/variants/disk_v1/mod.rs

```diff
@@ -241,13 +249,12 @@ where
         data_dir: path.parent().expect("always a parent"),
     })?;
 
-    let mut byte_size = 0;
-    let mut key_count = 0;
     let mut first_key = None;
     let mut last_key = None;
+    let mut state = BufferState::default();
     for (k, v) in db.iter(ReadOptions::new()) {
-        byte_size += v.len() as u64;
-        key_count += 1;
+        state.total_bytes += v.len() as u64;
+        state.total_records += 1;
```

Comment:
I'm not a big fan of the `mut` state struct pattern. I'd rather see individual variables and construct the resulting state at the end, which also avoids the need for a `Default` here.

Reply:
I can I have no problem cleaning it up to fit that model.

For my own curiosity, though: what do you see as the downsides of the state struct being used mutably? Given that there's no way to freeze the struct, given visibility rules within the same module, it feels like the same difference regardless of how I construct it.

Reply:
It makes no real changes for _this_ struct, as all the data is simple integers. It's more a pattern issue to avoid as a general practice:
1. The variables have needlessly longer names (trivial).
2. It uses struct internals in a non-member function (relatively trivial)
3. With other data types, there could be the overhead of extra allocations, or `Drop` on early return, or other side effects.
4. Without the struct variables, this particular loop could be reduced to a `fold` adapter on the iterator (albeit somewhat complex given the four pieces of data being folded) to eliminate mutable state.

Reply:
Yeah, OK, I buy that and appreciate the thoughtfulness of the response.

This will be good information that we can hopefully add to a style guide, in the future, if I get around to writing the RFC... 😅